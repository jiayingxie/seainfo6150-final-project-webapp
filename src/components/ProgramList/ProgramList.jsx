import React, { Component } from 'react';
import ProgramCard from '../ProgramCard/ProgramCard.jsx';
import styles from './ProgramList.module.css';

export default class ProgramList extends Component {
    render() {
        const { department } = this.props;
        let programs = Object.values(department.programs);
        let displayContent = (
            <ul className={styles.container} >
                {programs.map(program => <ProgramCard key={program.programID} {...program} department={this.props.department.departmentID} />)}
            </ul>
        )

        return (
            <div>
                {displayContent}
            </div>
        )
    }
}
