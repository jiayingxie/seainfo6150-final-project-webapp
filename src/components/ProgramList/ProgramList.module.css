.container {
    border-radius: 10px;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: 20px;
    font-family: Arial, Helvetica, sans-serif;
    list-style-type: none;
    padding: 15px;
}

@supports (display: grid) {
    .container {
      grid-template-columns: 1fr;
      font-size: 18px;
    }
  
    @media (min-width: 600px) {
      .container {
        grid-template-columns: 1fr 1fr;
        font-size: 20px;
      }
    }
  
    @media (min-width: 900px) {
      .container {
        grid-template-columns: 1fr 1fr 1fr;
        font-size: 22px;
      }
    }

    @media (min-width: 1200px) {
      .container {
        grid-template-columns: repeat(4, 1fr);
        font-size: 24px;
      }
    }
}