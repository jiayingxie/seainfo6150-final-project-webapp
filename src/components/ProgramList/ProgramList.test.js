import React from "react";
import ProgramList from "./ProgramList";

describe("ProgramList tests", () => {
    const mul = "multidisciplinary";
    // the method is to get test data
    const fetchData = async () => {
        const response = await fetch(`https://demo2247515.mockable.io/engineering`);
        const responseJson = await response.json();
        return Object.values(responseJson);
    };
    it("renders correctly when passed a department prop", async () => {
        const fetchedData = await fetchData();
        const department = fetchedData.find(dep => { return dep.departmentID === mul });
        const { container } = render(<ProgramList department={department} />);
        expect(container).toMatchSnapshot();
    });
});