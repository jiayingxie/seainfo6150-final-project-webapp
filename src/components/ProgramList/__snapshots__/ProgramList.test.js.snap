// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramList tests renders correctly when passed a department prop 1`] = `
<div>
  <div>
    <ul
      class="container"
    >
      <li
        class="container"
      >
        <div
          class="card"
        >
          <h2>
            CSECS
          </h2>
          <h4
            class="link"
          >
            <a
              href="/engineering/multidisciplinary/CSECS"
            >
              Computer Systems Engineering with Concentration in Software Design Engineering
            </a>
          </h4>
        </div>
      </li>
      <li
        class="container"
      >
        <div
          class="card"
        >
          <h2>
            IS
          </h2>
          <h4
            class="link"
          >
            <a
              href="/engineering/multidisciplinary/IS"
            >
              Information Systems
            </a>
          </h4>
        </div>
      </li>
      <li
        class="container"
      >
        <div
          class="card"
        >
          <h2>
            DAMG
          </h2>
          <h4
            class="link"
          >
            <a
              href="/engineering/multidisciplinary/DAMG"
            >
              Data Architecture and Management
            </a>
          </h4>
        </div>
      </li>
      <li
        class="container"
      >
        <div
          class="card"
        >
          <h2>
            CPS
          </h2>
          <h4
            class="link"
          >
            <a
              href="/engineering/multidisciplinary/CPS"
            >
              Cyber-Physical Systems, MS
            </a>
          </h4>
        </div>
      </li>
      <li
        class="container"
      >
        <div
          class="card"
        >
          <h2>
            SES
          </h2>
          <h4
            class="link"
          >
            <a
              href="/engineering/multidisciplinary/SES"
            >
              Software Engineering Systems, MS
            </a>
          </h4>
        </div>
      </li>
      <li
        class="container"
      >
        <div
          class="card"
        >
          <h2>
            TN
          </h2>
          <h4
            class="link"
          >
            <a
              href="/engineering/multidisciplinary/TN"
            >
              Telecommunication Networks, MS
            </a>
          </h4>
        </div>
      </li>
      <li
        class="container"
      >
        <div
          class="card"
        >
          <h2>
            BSCE
          </h2>
          <h4
            class="link"
          >
            <a
              href="/engineering/multidisciplinary/BSCE"
            >
              Blockchain and Smart Contract Engineering, Graduate Certificate
            </a>
          </h4>
        </div>
      </li>
      <li
        class="container"
      >
        <div
          class="card"
        >
          <h2>
            BWS
          </h2>
          <h4
            class="link"
          >
            <a
              href="/engineering/multidisciplinary/BWS"
            >
              Broadband Wireless Systems, Graduate Certificate
            </a>
          </h4>
        </div>
      </li>
      <li
        class="container"
      >
        <div
          class="card"
        >
          <h2>
            CSE
          </h2>
          <h4
            class="link"
          >
            <a
              href="/engineering/multidisciplinary/CSE"
            >
              Computer Systems Engineering, Graduate Certificate
            </a>
          </h4>
        </div>
      </li>
      <li
        class="container"
      >
        <div
          class="card"
        >
          <h2>
            EL
          </h2>
          <h4
            class="link"
          >
            <a
              href="/engineering/multidisciplinary/EL"
            >
              Engineering Leadership, Graduate Certificate
            </a>
          </h4>
        </div>
      </li>
      <li
        class="container"
      >
        <div
          class="card"
        >
          <h2>
            IPTS
          </h2>
          <h4
            class="link"
          >
            <a
              href="/engineering/multidisciplinary/IPTS"
            >
              IP Telephony Systems, Graduate Certificate
            </a>
          </h4>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
